src/logging.c

```diff
@@ -72,6 +103,44 @@ void rcutils_log(
   }
 }
 
+/*!
+  RCUTILS_LOGGING_ENSURE_LARGE_ENOUGH_BUFFER(
+   n, output_buffer_size, allocator, output_buffer, static_output_buffer)
+  Macro for (re)allocating a dynamic buffer if the ouput buffer doesn't have enough space for an
+  additional n characters and null terminator. Whether to allocate or re-allocate is determined by
+  if the output_buffer points to the original static_output_buffer or not. If a new buffer is
+  allocated it will be padded with NUL.
+*/
+
+#define RCUTILS_LOGGING_ENSURE_LARGE_ENOUGH_BUFFER( \
+    n, output_buffer_size, allocator, output_buffer, static_output_buffer) \
+  size_t required_output_buffer_size = strlen(output_buffer) + n + 1; \
+  if (required_output_buffer_size >= output_buffer_size) { \
+    do { \
+      output_buffer_size *= 2; \
+    } while (required_output_buffer_size >= output_buffer_size); \
+    if (output_buffer == static_output_buffer) { \
+      void * dynamic_output_buffer = allocator.allocate(output_buffer_size, allocator.state); \
+      if (!dynamic_output_buffer) { \
+        fprintf(stderr, "failed to allocate buffer for logging output\n"); \
+        goto cleanup; \
+      } \
+      memset(dynamic_output_buffer, '\0', output_buffer_size); \
+      memcpy(dynamic_output_buffer, output_buffer, strlen(output_buffer)); \
+      output_buffer = (char *)dynamic_output_buffer; \
+    } else { \
+      void * new_dynamic_output_buffer = (char *)allocator.reallocate( \
```

Comment:
Could this be of type `char *`?

Reply:
TBH in general I was copying https://github.com/ros2/rcutils/pull/23/files#diff-f909d0b1a298531f76132943dc55f86aR124 and went a bit off-script. In cd67099b586950a6e32544d6e122bae65b2e2a6c I removed this cast to match the line I just linked to, but I'm also happy to update all of them to not use `void *`